contracts-periphery/test/UniswapWithdrawHook.t.sol

```diff
@@ -0,0 +1,131 @@
+  ) external;
+}
+
+contract UniswapWithdrawHookTest is DSTestPlus {
+  using SafeERC20 for IERC20;
+  UniswapWithdrawHook withdrawHook;
+
+  IUmbra umbraContract;
+  ISwapRouter swapRouter;
+  IERC20 dai;
+
+  uint256 toll;
+  bytes32 pkx = "pkx";
+  bytes32 ciphertext = "ciphertext";
+
+  address alice = address(0x202204);
+  address bob = address(0x202205);
+  address feeReceiver = address(0x202206);
+  address constant umbra = 0xFb2dc580Eed955B528407b4d36FfaFe3da685401;
+
+  address public constant DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+  address public constant WETH9 = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+  address public constant Router = 0x68b3465833fb72A70ecDF485E0e4C7bD8665Fc45;
+  uint24 poolFee = 3000;
+
+  function setUp() public {
+    vm.etch(umbra, (deployCode("test/utils/Umbra.json", bytes(abi.encode(0, address(this), address(this))))).code);
+    umbraContract = IUmbra(address(umbra));
+    swapRouter = ISwapRouter(Router);
+    withdrawHook = new UniswapWithdrawHook(ISwapRouter(swapRouter));
+    dai = IERC20(DAI);
+    //Owner approves tokens
+    withdrawHook.approveToken(dai);
+    deal(address(DAI), address(this), 1e7 ether);
+  }
+
+  function testFuzz_HookTest(uint256 amount, uint256 swapAmount) public {
+    amount=bound(amount, 10 ether, 10e20);
+    swapAmount=bound(swapAmount, 10 ether, amount);
+    dai.approve(address(umbraContract), amount);
+
+
+    umbraContract.sendToken{value: toll}(
+    address(alice),
+    address(DAI),
+    amount,
+    pkx,
+    ciphertext
+    );
+
+    vm.startPrank(alice); // Withdraw as Alice
+    address destinationAddr = bob;
+    uint256 minOut;
+    poolFee = 3000;
+    IUmbraHookReceiver receiver = IUmbraHookReceiver(address(withdrawHook));
+
+    bytes memory _path = abi.encodePacked(address(DAI), poolFee, WETH9);
+
+    ISwapRouter.ExactInputParams memory params;
+    params =
+      ISwapRouter.ExactInputParams({
+        path: _path,
+        recipient : address(swapRouter),
+        amountIn : swapAmount,
+        amountOutMinimum: minOut
+      });
+
+    bytes[] memory multicallData = new bytes[](2);
+    // multicallData[0] = abi.encodeWithSelector(swapRouter.exactInput.selector, params);
+    multicallData[0] = abi.encodeCall(swapRouter.exactInput, params);
+
+    // params.amountOutMinimum might need to be a different value
+    multicallData[1] = abi.encodeCall(swapRouter.unwrapWETH9WithFee, (params.amountOutMinimum, destinationAddr, 1, feeReceiver));
+
+    bytes memory data = abi.encode(destinationAddr, multicallData);
+
+    vm.expectCall(umbra, abi.encodeWithSelector(umbraContract.withdrawTokenAndCall.selector));
+    vm.expectCall(address(withdrawHook), abi.encodeWithSelector(withdrawHook.tokensWithdrawn.selector));
+
+    umbraContract.withdrawTokenAndCall(address(withdrawHook), address(DAI), receiver, data);
+
+    assertEq(IERC20(DAI).balanceOf(address(bob)), amount-swapAmount);
+    assertTrue(address(destinationAddr).balance > minOut);
```

Comment:
Can we add tighter assertions around the input/output amounts and ensure fee is properly sent to the fee recipient? Since we fork some a specific block, the amounts will be constant, meaning you can run the test once to see the results, sanity check them, and if they make sense hardcode them. We could fuzz over the recipient, fee amount, and fee receiver too

Reply:
Added this for now, but will try to add more fuzz https://github.com/ScopeLift/umbra-protocol/pull/341/commits/1410f85e728f1e0051733f921574124a022e85c5

Reply:
Added stricter assertions in https://github.com/ScopeLift/umbra-protocol/pull/341/commits/da3ef52c8c954203fe1cb68c85b3207c5436ac90

Reply:
Two few comments here:
1. Why do we have two separate tests that generate random addresses in a loop (one only runs for one iteration), as opposed to just using `testFuzz_HookTest` directly which will fuzz over the fee parameters?
2. Instead of `assertTrue(x > y)` it's more clear IMO to use `assertGe` / `assertGt`